File: 백준/Silver/1182. 부분수열의 합/부분수열의 합.py
from itertools import combinations

x, y = map(int, input().split())
mList = list(map(int, input().split()))

count = 0

for i in range(1, x+1):
    now = list(combinations(mList, i))
    for n in now:
        sum = 0
        for j in range(i):
            sum += n[j]

        if sum == y:
            count += 1
print(count)


# [('1', '2'), ('1', '3'), ('1', '4'), ('1', '5'), ('2', '3'), ('2', '4'), ('2', '5'), ('3', '4'), ('3', '5'), ('4', '5')]

File: 백준/Silver/1182. 부분수열의 합/부분수열의 합.js
function recur(start, sum){
    if(targetNum===sum){
        answer++;
    }
    if(start===arr.length-1){
        return;
    }

    for(let i=start+1; i<arr.length;i++){
        recur(i, sum+arr[i]);
    }
}

function solution(lines){
    targetNum=parseInt(lines[0].split(' ')[1]);
    arr=lines[1].split(' ').map(el=>parseInt(el));

    for(let i=0; i<arr.length; i++){
        recur(i, arr[i])
    }

    console.log(answer);
}
let targetNum=0;
let answer=0;
let arr=[];

const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});
let list = [];
let count=0;


rl.on('line', function(line) {
        list.push(line);
        count++;

        if(count===2){
            rl.close();
        }
}).on("close", function() {
    //for(let i=0; i<list.length; i++){
        solution(list);
    //}
    process.exit();
});
